src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -80,10 +80,17 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '' or '$(TargetFrameworkVersion)' == ''">
     <_UnsupportedTargetFrameworkError>true</_UnsupportedTargetFrameworkError>
   </PropertyGroup>
-  <Target Name="_CheckForUnsupportedTargetFramework" BeforeTargets="_CheckForInvalidConfigurationAndPlatform">
-    <NETSdkError Condition="'$(_UnsupportedTargetFrameworkError)' == 'true'" 
+  <Target Name="_CheckForUnsupportedTargetFramework"
+          BeforeTargets="_CheckForInvalidConfigurationAndPlatform;Restore;CollectPackageReferences"
```

Comment:
IIRC, this won't run before restore unless it's the root project. We use a private target (_GenerateRestoreSpec or something, search for it) with @emgarten's approval elsewhere.